### DISCUSSION TITLE
Dicas Sobre React Native. Formas de praticar a codar e sobre testes.

### DISCUSSION THREAD

**Discussion by Kapeite1** [frontendbr/forum#1926] [Discussão]
###Formas de praticar

Fala devs, me considero em um estágio de conhecimento com a biblioteca React Native intermediária, já desenvolvi alguns apps utilizando o mesmo, como por exemplo um app para delivery de uma lanchonete e outro para controle de financias. 
O problema é que me sinto estagnado, não estou sabendo praticar, ter ideias para conseguir desenvolver e me esbarrar com problemas, gostaria de dicas e assuntos para estudar, aprofundar e finalmente conseguir entrar no mercado utilizando React Native.

###Testes

Os testes para mim é algo novo, ainda não peguei para estudar, gostaria de dicas e qual escolher, alguem poderia me ajudar?  Jest, mocha, chai, teste unitario, tdd, de integração, end to end? 

Obrigado a todos pela colaboração.

**Comment by victorferreira** [comment]
Se inglês não for um problema, o canal do [William Candillon](https://www.youtube.com/user/wcandill) tem muita coisa bem legal sendo implementada com RN.

**Comment by ildaneta** [comment]
Olá @Kapeite1, tudo bem?

Como dica, acho que o principal é ter conhecimentos de javascript e noções visuais, como flexbox por exemplo.
Quando você se refere a estar estagnado, é em relação a não conseguir evoluir tecnicamente?

Em relação a testes, é legal olhar aqueles que são mais utilizados pela comunidade e analisar os prós e contras de cada um, e para iniciar, acho que poderia dar uma olhada no Jest e React Testing Library.
